Solution1/ProjectClock.BussinesLogic/Services/RaportServices/RaportServices.cs

```diff
@@ -0,0 +1,386 @@
+            {
+
+               StartData = new DateTime(Random.Shared.Next(1950, 2011),
+                Random.Shared.Next(1, 13),
+                Random.Shared.Next(1, 28))
+            },
+            new GetNameAndIdAllUsersDto()
+            {
+
+                StartData = new DateTime(Random.Shared.Next(1950, 2011),
+                Random.Shared.Next(1, 13),
+                Random.Shared.Next(1, 28))
+            },
+            new GetNameAndIdAllUsersDto()
+            {
+
+               StartData = new DateTime(Random.Shared.Next(1950, 2011),
+                Random.Shared.Next(1, 13),
+                Random.Shared.Next(1, 28))
+            },
+        };
+
+
+
+
+
+
+
+
+
+
+
+
+        }
+
+        public RaportServices(ProjectClockDbContext projectClockDbContext, IMapper mapper, IUserServices userService, IProjectServices projectService, IWorkingTimeServices workingTimeServices, ILogger<RaportServices> logger, ProjectClockDbContext dbContext)
+        {
+            _projectClockDbContext = projectClockDbContext;
+            _mapper = mapper;
+
+			_userService = userService;
+            _projectService = projectService;
+            _workingTimeServices = workingTimeServices;
+            _logger = logger;
+
+            _dbContext = dbContext;
+
+
+
+
+
+        }
+
+        public async Task<bool> AddProjectToOrganization(int organizationId, Project project)
+        {
+
+            var organization = await _projectClockDbContext.Organizations.FindAsync(organizationId);
+
+            if (organization == null)
+            {
+                return false;
+            }
+
+
+            organization.Projects.Add(project);
+            await _projectClockDbContext.SaveChangesAsync();
+
+            return true;
+        }
+
+        public async Task<List<WorkingTime>> GetUserWorkingTimesAfterUserId(int userId)
+        {
+
+            var list = await _projectClockDbContext.WorkingTimes
+
+                 .Where(e => e.UserId == userId)
+                 .ToListAsync();
+
+
+            return list;
+        }
+
+
+        public async Task<List<Project>> GetAllProjects()
+        {
+
+            var list = await _projectClockDbContext.Projects.ToListAsync();
+
+
+            return list;
+        }
+
+
+        public void CreateParticipant(GetNameAndIdAllUsersDto participant)
+        {
+            
+            _participants.Add(participant);
+        }
+
+       
+        public async Task<List<WorkingTime>> GetAll()
+        {
+
+            return await _projectClockDbContext.WorkingTimes.ToListAsync();
+
+        }
+
+
+		public async Task<IEnumerable<WorkingTimeDto>> GetUserAllWorkingTimes(int userId)
+		{
+			var list = await _projectClockDbContext.WorkingTimes
+
+
+				.Where(e => e.UserId == userId)
+				.Include(wt => wt.Project)
+				.Include(wt => wt.User)
+				.ToListAsync();
+
+			var dtos = _mapper.Map<IEnumerable<WorkingTimeDto>>(list);
+
+			return dtos;
+		}
+ 
```

Comment:
Nie rozumiem trochę po co przepisujesz metody z innych serwisów,

Reply:
Teraz akurat przepisałem bo to mały projekt, jak bedzie jekiś duży projekt i tak bedę musial napisać swoją zamiast szukać odpowiedniej pośród setek innych metod i może być tak, że się zduplikują. To po pierwsze a po drugie lubię być "odcięty od reszty projektu".

Reply:
ale tak się nie robi :(

Reply:
Tomasz ja to pisałem wszystko 3 - 4 dni, zrobię to tak jak należy ale potrzebuje trochę czasu. To jest narazie szkic jak już będę miał ogarnięte podstawy doszlifuję wszytko. Tymek chciał aby było coś na zakończenie i wy też, ja myślałem że na 11 maja jest koniec S2